<a href="https://colab.research.google.com/github/KimberlySalazarB/Actividad1/blob/main/Actividad1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [23]:
!head data.tsv

order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 


In [24]:
!tail data.tsv

1831	1	Chips	NULL	$2.15 
1831	1	Bottled Water	NULL	$1.50 
1832	1	Chicken Soft Tacos	[Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]	$8.75 
1832	1	Chips and Guacamole	NULL	$4.45 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Guacamole, Lettuce]]	$11.25 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Lettuce]]	$8.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Lettuce]]	$8.75 



**1.- Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?**

*Columnas:*

order_id: son los ID de las ordenes o pedidos que se hicieron.

quantity: es la cantidad del producto que se esta pidiendo.

item_name: nombre del producto.

choice_description: descripcion del producto.

item_price: es el precio del producto

*Filas:*

Son los datos de cada variable. Asimismo, en algunas se  puede encontrar con la palabra NULL eso quiere decir que esta vacio.

**2.-¿Cuántos pedidos parece haber?**

In [11]:
!awk -F'\t' '{print $1}' data.tsv | sort |uniq |wc -l

1836


Son 1836 pedidos

**3.-¿Cuántas líneas hay en este archivo?**

Hay 4624 filas

In [12]:
!wc -l data.tsv

4624 data.tsv


**4.-¿Qué burrito es más popular, steak o chicken?**

El Burrito Chicken porque tiene 553.

In [15]:
!echo 'Chicken Burrito:'
! grep -i 'Chicken Burrito' data.tsv |wc -l

Chicken Burrito:
553


In [16]:
!echo 'Steak Burrito:'
! grep -i 'Steak Burrito' data.tsv |wc -l

Steak Burrito:
368


**5.-¿Los burritos de pollo suelen tener frijoles negros o frijoles pintos?**

Ambos frijoles, también en algunos suelen tener fijoles tintos o bien fijoñes pintos

In [18]:
!echo 'Pinto Beans:'
! grep -i 'Chicken Burrito' data.tsv |grep -i 'Pinto Beans'|wc -l

Pinto Beans:
105


In [19]:
!echo 'Black Beans:'
! grep -i 'Chicken Burrito' data.tsv |grep -i 'Black Beans'|wc -l

Black Beans:
282


**6.- Usa la línea de comandos para descubrir algo interesante sobre los datos dados.**

Se utiliza para imprimir 10 primeras lineas  de coincidencia con la palabra que se esta buscando que en este caso es Bottled Water.

In [47]:
!grep -m 10 "Bottled Water" data.tsv

17	1	Bottled Water	NULL	$1.09 
38	1	Bottled Water	NULL	$1.09 
138	1	Bottled Water	NULL	$1.09 
143	1	Bottled Water	NULL	$1.50 
163	1	Bottled Water	NULL	$1.50 
203	1	Bottled Water	NULL	$1.50 
225	2	Bottled Water	NULL	$3.00 
244	1	Bottled Water	NULL	$1.50 
270	1	Bottled Water	NULL	$1.50 
288	1	Bottled Water	NULL	$1.09 


Con este comando modificamos la palabra NULL por vacio  de las 10 primeras lineas de coincidencia con la palabra que se esta buscando que en este caso es Bottled Water.

In [53]:
!sed 's/NULL/vacio/g' data.tsv | grep -m 10 "Bottled Water"

17	1	Bottled Water	vacio	$1.09 
38	1	Bottled Water	vacio	$1.09 
138	1	Bottled Water	vacio	$1.09 
143	1	Bottled Water	vacio	$1.50 
163	1	Bottled Water	vacio	$1.50 
203	1	Bottled Water	vacio	$1.50 
225	2	Bottled Water	vacio	$3.00 
244	1	Bottled Water	vacio	$1.50 
270	1	Bottled Water	vacio	$1.50 
288	1	Bottled Water	vacio	$1.09 
